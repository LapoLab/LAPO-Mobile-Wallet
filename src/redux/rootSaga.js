import { all } from 'redux-saga/effects';
import commonsSagas from '../commons/sagas';
import recoveryWordsSagas from '../modules/RecoveryWords/sagas';
import sendSagas from '../modules/Send/sagas';
import paperWordsConfirm from '../modules/PaperWordsConfirm/sagas';
import dashboard from '../modules/Dashboard/sagas';
import settings from '../modules/Settings/sagas';
import paperWords from '../modules/PaperWords/sagas';
import transactionsDetails from '../modules/TransactionDetails/sagas';
import chooseContact from '../modules/ChooseContact/sagas';
import splashScreen from '../modules/SplashScreen/sagas';
import switchContacts from '../modules/SwitchContacts/sagas';
import receive from '../modules/Receive/sagas';

// eslint-disable-next-line
const getListeners = (...args) => args.reduce((acc, nextArg) => [...acc, ...Object.values(nextArg).map(func => func())], []);

export default function* rootSaga() {
  yield all(
    getListeners(
      commonsSagas,
      recoveryWordsSagas,
      sendSagas,
      paperWordsConfirm,
      dashboard,
      settings,
      paperWords,
      transactionsDetails,
      chooseContact,
      splashScreen,
      switchContacts,
      receive,
    ),
  );
}
