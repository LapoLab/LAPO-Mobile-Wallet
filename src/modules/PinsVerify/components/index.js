/**
 * @flow
 */

import React, { Component } from 'react';
import {
  Text,
  View,
  SafeAreaView,
  NativeModules,
  Platform,
  BackHandler,
  AppState,
} from 'react-native';
import { NavigationActions } from 'react-navigation';
import LinearGradient from 'react-native-linear-gradient';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import LapoCoreAndroid from 'lapo-core';
import * as actions from '../../../commons/actions';
import * as actionsAlert from '../../Alert/actions';
import styles from './styles';
import Pis from '../../../commons/components/Pins';
import SpinnerPage from '../../../commons/components/SpinnerPage';

const LapoCore = Platform.OS === 'ios' ? NativeModules.RNLapo : LapoCoreAndroid;

type Props = {
  navigation: {
    openDrawer: Function,
    dispatch: Function,
    navigate: Function,
    state: { params: { currentRoute: string } },
  },
  pins: string,
  isStartGetInfo: boolean,
  loadWallet: Function,
  showAlert: Function,
  hideAlert: Function,
  setShowSpinner: Function,
};
type State = {
  pinsVerify: string,
  appState: string,
};
class PinsVerify extends Component<Props, State> {
  state = {
    pinsVerify: '',
    appState: AppState.currentState,
  };

  componentDidMount() {
    BackHandler.addEventListener('hardwareBackPress', this.handleBack);
    AppState.addEventListener('change', this.handleAppStateChange);
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress', this.handleBack);
    AppState.removeEventListener('change', this.handleAppStateChange);
  }

  handleAppStateChange = (nextAppState: string) => {
    const { appState } = this.state;
    if (nextAppState.match(/inactive|background/) && appState === 'active') {
      this.setState({ pinsVerify: '' });
    }
    this.setState({ appState: nextAppState });
  };

  handleBack = () => {
    LapoCore.disconnect();
    BackHandler.exitApp();
    return true;
  };

  navigateToScreen = route => () => {
    const {
      navigation: { dispatch },
    } = this.props;
    const navigateAction = NavigationActions.navigate({
      routeName: route,
    });
    dispatch(navigateAction);
  };

  handleChangePin = async (value) => {
    const {
      pins: pinsOriginal,
      loadWallet,
      navigation,
      showAlert,
      hideAlert,
      setShowSpinner,
      isStartGetInfo,
    } = this.props;
    const { pinsVerify } = this.state;

    if (value === 'delete') {
      this.setState({ pinsVerify: pinsVerify.slice(0, pinsVerify.length - 1) });
      return;
    }

    const curPins = pinsVerify.length < 6 ? `${pinsVerify}${value}` : pinsVerify;

    this.setState({ pinsVerify: curPins });

    if (curPins.length === 6 && pinsOriginal === curPins) {
      setShowSpinner(true);
      if (isStartGetInfo) {
        navigation.navigate(
          navigation.state.params.currentRoute
            && navigation.state.params.currentRoute !== 'PinsVerify'
            ? navigation.state.params.currentRoute
            : 'WorkStack',
        );
        setShowSpinner(false);
      } else {
        loadWallet();
      }
    } else if (curPins.length === 6) {
      showAlert(
        'Error',
        'Wrong password',
        // eslint-disable-next-line
        [
          {
            text: 'OK',
            onPress: () => {
              this.setState({ pinsVerify: '' });
              hideAlert();
            },
          },
        ],
        {
          cancelable: false,
        },
      );
    }
  };

  render() {
    const { pinsVerify } = this.state;
    return (
      <LinearGradient
        start={{ x: -0.1, y: -0.1 }}
        end={{ x: 1.2, y: 1.2 }}
        locations={[0, 1]}
        colors={['#4183BD', '#093B0C']}
        style={{ flex: 1 }}
      >
        <SafeAreaView style={styles.container}>
          <View style={styles.header}>
            <Text style={styles.titleHeader}>Unlock your wallet</Text>
          </View>

          <View style={styles.pins}>
            <Pis pins={pinsVerify} onChange={this.handleChangePin} />
          </View>
        </SafeAreaView>
        <SpinnerPage />
      </LinearGradient>
    );
  }
}

const mapStateToProps = state => ({
  pins: state.pins.pins,
  isStartGetInfo: state.dashboard.isStartGetInfo,
});

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    loadWallet: actions.loadWallet,
    showAlert: actionsAlert.showAlert,
    hideAlert: actionsAlert.hideAlert,
    setShowSpinner: actions.setShowSpinner,
  },
  dispatch,
);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PinsVerify);
