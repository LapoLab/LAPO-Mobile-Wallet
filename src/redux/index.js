import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import rootReducer from './rootReducer';
import rootSagas from './rootSaga';

const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
  blacklist: [
    'modals',
    'dashboard',
    'paperWords',
    'chooseContact',
    'send',
    'transactionsDetails',
    'alert',
    'pinsRecovery',
  ],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const sagaMiddleware = createSagaMiddleware();
// eslint-disable-next-line no-undef
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(sagaMiddleware)),
);
sagaMiddleware.run(rootSagas);

export const persistor = persistStore(store);
