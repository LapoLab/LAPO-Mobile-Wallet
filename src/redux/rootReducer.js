import { combineReducers } from 'redux';

import wallet from '../commons/reducers/wallet';
import pins from '../commons/reducers/pins';
import pinsRecovery from '../commons/reducers/pinsRecovery';
import modals from '../commons/reducers/modals';
import settings from '../commons/reducers/settings';
import notify from '../modules/Notify/reducers';
import paperConfirm from '../modules/PaperWordsConfirm/reducer';
import dashboard from '../modules/Dashboard/reducers';
import paperWords from '../modules/PaperWords/reducers';
import chooseContact from '../modules/ChooseContact/reducers';
import send from '../modules/Send/reducers';
import transactionsDetails from '../modules/TransactionDetails/reducer';
import alert from '../modules/Alert/reducers';

export default combineReducers({
  wallet,
  pins,
  pinsRecovery,
  notify,
  modals,
  settings,
  paperConfirm,
  dashboard,
  paperWords,
  chooseContact,
  send,
  transactionsDetails,
  alert,
});
