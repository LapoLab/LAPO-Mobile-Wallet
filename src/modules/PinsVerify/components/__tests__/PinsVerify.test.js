// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../redux';
import PinsVerify from '../index';

describe('PinsVerify page', () => {
  test('renders snapshots', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PinsVerify />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
