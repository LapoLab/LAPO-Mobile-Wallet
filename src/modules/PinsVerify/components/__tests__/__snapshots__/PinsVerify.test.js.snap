// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinsVerify page renders snapshots 1`] = `
<BVLinearGradient
  colors={
    Array [
      -12483651,
      -16172276,
    ]
  }
  endPoint={
    Object {
      "x": 1.2,
      "y": 1.2,
    }
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  startPoint={
    Object {
      "x": -0.1,
      "y": -0.1,
    }
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "flex-start",
          "marginTop": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "white",
            "flex": 1,
            "fontFamily": "Lato-Regular",
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        Unlock your wallet
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginTop": 80,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "paddingBottom": 10,
            "paddingHorizontal": "10%",
            "paddingdVertical": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
                "paddingVertical": 10,
                "width": 180,
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "#4183BD",
                  "borderRadius": 6,
                  "borderWidth": 1,
                  "height": 12,
                  "width": 12,
                }
              }
            />
            <View
              style={
                Object {
                  "borderColor": "#4183BD",
                  "borderRadius": 6,
                  "borderWidth": 1,
                  "height": 12,
                  "width": 12,
                }
              }
            />
            <View
              style={
                Object {
                  "borderColor": "#4183BD",
                  "borderRadius": 6,
                  "borderWidth": 1,
                  "height": 12,
                  "width": 12,
                }
              }
            />
            <View
              style={
                Object {
                  "borderColor": "#4183BD",
                  "borderRadius": 6,
                  "borderWidth": 1,
                  "height": 12,
                  "width": 12,
                }
              }
            />
            <View
              style={
                Object {
                  "borderColor": "#4183BD",
                  "borderRadius": 6,
                  "borderWidth": 1,
                  "height": 12,
                  "width": 12,
                }
              }
            />
            <View
              style={
                Object {
                  "borderColor": "#4183BD",
                  "borderRadius": 6,
                  "borderWidth": 1,
                  "height": 12,
                  "width": 12,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(93,112,241, 0.2)",
                  "borderRadius": 48,
                  "height": 96,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 96,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#4183BD",
                    "borderRadius": 40,
                    "height": 80,
                    "justifyContent": "center",
                    "width": 80,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Lato-Bold",
                      "fontSize": 20,
                    }
                  }
                >
                  1
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(93,112,241, 0.2)",
                  "borderRadius": 48,
                  "height": 96,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 96,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#4183BD",
                    "borderRadius": 40,
                    "height": 80,
                    "justifyContent": "center",
                    "width": 80,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Lato-Bold",
                      "fontSize": 20,
                    }
                  }
                >
                  2
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(93,112,241, 0.2)",
                  "borderRadius": 48,
                  "height": 96,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 96,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#4183BD",
                    "borderRadius": 40,
                    "height": 80,
                    "justifyContent": "center",
                    "width": 80,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Lato-Bold",
                      "fontSize": 20,
                    }
                  }
                >
                  3
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(93,112,241, 0.2)",
                  "borderRadius": 48,
                  "height": 96,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 96,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#4183BD",
                    "borderRadius": 40,
                    "height": 80,
                    "justifyContent": "center",
                    "width": 80,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Lato-Bold",
                      "fontSize": 20,
                    }
                  }
                >
                  4
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(93,112,241, 0.2)",
                  "borderRadius": 48,
                  "height": 96,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 96,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#4183BD",
                    "borderRadius": 40,
                    "height": 80,
                    "justifyContent": "center",
                    "width": 80,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Lato-Bold",
                      "fontSize": 20,
                    }
                  }
                >
                  5
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(93,112,241, 0.2)",
                  "borderRadius": 48,
                  "height": 96,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 96,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#4183BD",
                    "borderRadius": 40,
                    "height": 80,
                    "justifyContent": "center",
                    "width": 80,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Lato-Bold",
                      "fontSize": 20,
                    }
                  }
                >
                  6
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(93,112,241, 0.2)",
                  "borderRadius": 48,
                  "height": 96,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 96,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#4183BD",
                    "borderRadius": 40,
                    "height": 80,
                    "justifyContent": "center",
                    "width": 80,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Lato-Bold",
                      "fontSize": 20,
                    }
                  }
                >
                  7
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(93,112,241, 0.2)",
                  "borderRadius": 48,
                  "height": 96,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 96,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#4183BD",
                    "borderRadius": 40,
                    "height": 80,
                    "justifyContent": "center",
                    "width": 80,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Lato-Bold",
                      "fontSize": 20,
                    }
                  }
                >
                  8
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(93,112,241, 0.2)",
                  "borderRadius": 48,
                  "height": 96,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 96,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#4183BD",
                    "borderRadius": 40,
                    "height": 80,
                    "justifyContent": "center",
                    "width": 80,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Lato-Bold",
                      "fontSize": 20,
                    }
                  }
                >
                  9
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingVertical": 20,
                "textAlign": "center",
                "width": 150,
              }
            }
          />
          <View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(93,112,241, 0.2)",
                  "borderRadius": 48,
                  "height": 96,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 96,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#4183BD",
                    "borderRadius": 40,
                    "height": 80,
                    "justifyContent": "center",
                    "width": 80,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontFamily": "Lato-Bold",
                      "fontSize": 20,
                    }
                  }
                >
                  0
                </Text>
              </View>
            </View>
          </View>
          <View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "opacity": 1,
                  "paddingVertical": 20,
                  "textAlign": "center",
                  "width": 150,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#D5D9F3",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 14,
                    "marginRight": 40,
                  }
                }
              >
                DELETE
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
